src/openrct2/rct12/SawyerChunkReader.cpp

```diff
@@ -302,25 +289,7 @@ size_t SawyerChunkReader::DecodeChunkRotate(void* dst, size_t dstCapacity, const
     return srcLength;
 }
 
-void* SawyerChunkReader::AllocateLargeTempBuffer()
-{
-#ifdef __USE_HEAP_ALLOC__
-    auto buffer = HeapAlloc(GetProcessHeap(), 0, MAX_UNCOMPRESSED_CHUNK_SIZE);
-#else
-    auto buffer = std::malloc(MAX_UNCOMPRESSED_CHUNK_SIZE);
-#endif
-    if (buffer == nullptr)
-    {
-        throw std::runtime_error("Unable to allocate large temporary buffer.");
-    }
-    return buffer;
-}
-
-void SawyerChunkReader::FreeLargeTempBuffer(void* buffer)
+std::unique_ptr<uint8_t[]> SawyerChunkReader::AllocateLargeTempBuffer()
```

Comment:
Do we really need this function?

Reply:
Strictly speaking, no you could in line any function of your choosing.

Reply:
I don't think we require this function to just create a new unique_ptr is what I'm saying.